README.md

```diff
@@ -17,29 +17,97 @@
 
 ## Installation
 
-Cocotb can be installed by running `pip install cocotb`.
+Cocotb requires:
 
-## Quickstart
+- Python 3.5+
+- A C++ compiler
+- An HDL simulator (such as [Icarus Verilog](http://iverilog.icarus.com/))
 
-    # Install pre-requisites (waveform viewer optional)
-    sudo yum install -y iverilog python-devel gtkwave
+After installing these dependencies, the latest stable version of cocotb can be installed with pip.
 
-    # Checkout git repositories
-    git clone https://github.com/cocotb/cocotb.git
+```command
+pip install cocotb
+```
 
-    # Install cocotb
-    pip install ./cocotb
+For more details, including how to install a development version of cocotb, see [the documentation](https://docs.cocotb.org/en/latest/quickstart.html#pre-requisites).
 
-    # Run the tests...
-    cd cocotb/examples/endian_swapper/tests
-    make
+## Usage
 
-    # View the waveform
-    gtkwave waveform.vcd
+As a first trivial introduction to cocotb, the following example "tests" a flip-flop.
 
+First, we need a hardware design which we can test. For this example, create a file `dff.sv` with SystemVerilog code for a simple [D flip-flop](https://en.wikipedia.org/wiki/Flip-flop_(electronics)#D_flip-flop). You could also use any other language a [cocotb-supported simulator](https://docs.cocotb.org/en/latest/introduction.html) understands, e.g. VHDL.
 
-## Tutorials and examples
+```systemverilog
+// dff.sv
+
+`timescale 1us/1ns
+
+module dff (
+    output logic q,
+    input logic clk, d
+);
+
+always @(posedge clk) begin
+    q <= d;
+end
+
+endmodule
+```
+
+An example of a simple randomized cocotb testbench:
+
+```python
+# test_dff.py
+
+import random
+import cocotb
+from cocotb.clock import Clock
+from cocotb.triggers import FallingEdge
+
+@cocotb.test()
+async def test_dff_simple(dut):
+    """ Test that d propagates to q """
+
+    clock = Clock(dut.clk, 10, units="us")  # Create a 10us period clock on port clk
+    cocotb.fork(clock.start())  # Start the clock
+
+    for i in range(10):
+        val = random.randint(0, 1)
+        dut.d <= val  # Assign the random value val to the input port d
+        await FallingEdge(dut.clk)
+        assert dut.q == val, "output q was incorrect on the {}th cycle".format(i)
+```
+
+A simple Makefile:
+
+```make
+# Makefile
+
+TOPLEVEL_LANG = verilog
+VERILOG_SOURCES = $(shell pwd)/dff.sv
+TOPLEVEL = dff
+MODULE = test_dff
+
+include $(shell cocotb-config --makefiles)/Makefile.inc
+include $(shell cocotb-config --makefiles)/Makefile.sim
+```
+
+In order to run the test with Icarus Verilog, execute:
+
+```command
+make SIM=icarus
+```
+
+[![asciicast](https://asciinema.org/a/316949.svg)](https://asciinema.org/a/316949)
```

Comment:
Do we want to rebuild this now that #1581 is in, or do we not care?

Reply:
Heh. If you really want. Need some ❤️

Reply:
Could edit with https://discourse.asciinema.org/t/can-i-edit-post-process-the-recording/25